www/tests/test_app.py

```diff
@@ -1,4 +1,11 @@
 import app
 
-def test_dummy():
+def test_sensor_init():
     assert len(app.binary_sensors) > 0
+
+def test_get_local_ip():
```

Comment:
To test all of these effectively, you'll need to use something like [responses](https://github.com/getsentry/responses). It's a way to mock up the response from the server so that you don't actually need a server for the tests. Then you can check through all of the error cases as well. I can help you with it at some point if you run into issues. Basically, you setup a response to a particular request and see how the function behaves.